static/app/components/events/interfaces/crashContent/exception/actionableItemsUtils.tsx

```diff
@@ -0,0 +1,275 @@
+  | FetchGenericError
+  | RestrictedIpError
+  | SecurityViolationError
+  | FutureTimestampError
+  | ClockDriftError
+  | PastTimestampError
+  | ValueTooLongError
+  | InvalidDataError
+  | InvalidEnvironmentError
+  | InvalidAttributeError;
+
+export function shouldErrorBeShown(error: EventErrorData, event: Event) {
+  if (
+    error.type === CocoaProcessingErrors.COCOA_INVALID_DATA &&
+    event.sdk?.name === 'sentry.cocoa' &&
+    error.data?.name === 'contexts.trace.sampled' &&
+    semverCompare(event.sdk?.version || '', '8.7.4') === -1
+  ) {
+    // The Cocoa SDK sends wrong values for contexts.trace.sampled before 8.7.4
+    return false;
+  }
+  return true;
+}
+
+function isDataMinified(str: string | null) {
+  if (!str) {
+    return false;
+  }
+
+  return !![...str.matchAll(MINIFIED_DATA_JAVA_EVENT_REGEX_MATCH)].length;
+}
+
+const hasThreadOrExceptionMinifiedFrameData = (
+  definedEvent: Event,
+  bestThread?: Thread
+) => {
+  if (!bestThread) {
+    const exceptionValues: Array<ExceptionValue> =
+      definedEvent.entries?.find(e => e.type === EntryType.EXCEPTION)?.data?.values ?? [];
+
+    return !!exceptionValues.find(exceptionValue =>
+      exceptionValue.stacktrace?.frames?.find(frame => isDataMinified(frame.module))
+    );
+  }
+
+  const threadExceptionValues = getThreadException(definedEvent, bestThread)?.values;
+
+  return !!(threadExceptionValues
+    ? threadExceptionValues.find(threadExceptionValue =>
+        threadExceptionValue.stacktrace?.frames?.find(frame =>
+          isDataMinified(frame.module)
+        )
+      )
+    : bestThread?.stacktrace?.frames?.find(frame => isDataMinified(frame.module)));
+};
+
+export const useFetchProguardMappingFiles = ({
+  event,
+  isShare,
+  project,
+}: {
+  event: Event;
+  isShare: boolean;
+  project: Project;
+}): {proguardErrors: EventErrorData[]; proguardErrorsLoading: boolean} => {
+  const organization = useOrganization();
+  const hasEventErrorsProGuardMissingMapping = event.errors?.find(
+    error => error.type === 'proguard_missing_mapping'
+  );
+
+  const debugImages = event.entries?.find(e => e.type === EntryType.DEBUGMETA)?.data
+    .images as undefined | Array<Image>;
+
+  // When debugImages contains a 'proguard' entry, it must always be only one entry
+  const proGuardImage = debugImages?.find(debugImage => debugImage?.type === 'proguard');
+
+  const proGuardImageUuid = proGuardImage?.uuid;
+
+  const shouldFetch =
+    defined(proGuardImageUuid) &&
+    event.platform === 'java' &&
+    !hasEventErrorsProGuardMissingMapping &&
+    !isShare;
+
+  const {
+    data: proguardMappingFiles,
+    isSuccess,
+    isLoading,
+  } = useApiQuery<DebugFile[]>(
+    [
+      `/projects/${organization.slug}/${project.slug}/files/dsyms/`,
+      {
+        query: {
+          query: proGuardImageUuid,
+          file_formats: 'proguard',
+        },
+      },
+    ],
+    {
+      staleTime: Infinity,
+      enabled: shouldFetch,
+    }
```

Comment:
do we expect this to return bad status codes often like 404 or 400? Should we set retries to 0